### ISSUE TITLE
Add `always-single-line` to `function-comma-space-after`

### ISSUE THREAD

**Issue by MadLittleMods** [stylelint/stylelint#405]
Add `always-single-line` to [`function-comma-space-after`](https://github.com/stylelint/stylelint/blob/master/src/rules/function-comma-space-after/README.md)

Use case:

```
.foo {
	background: linear-gradient(
		45deg,
		rgba(215, 215, 215, 1) 0%,
		rgba(215, 215, 215, 1) 25%,
		rgba(53, 53, 53, 1) 65%,
		rgba(53, 53, 53, 1) 100%
	);
}
```

**Comment by jeddy3** [comment]
Thanks to the discussion in #406, I've now got my head around.

Your use case will involve a little more than adding an `always-single-line` option to `function-comma-space-after`. We'll need to bring `function-comma-*` in line with the flexibility of [`media-query-list-comma-*`](https://github.com/stylelint/stylelint/blob/master/docs/rules.md#media-query-list)

This means:

1. The creation of two new rules: `function-comma-newline-before` and `function-comma-newline-after`. 2. The existing `function-comma-space-*` rules will need `never/always-single-line` options added to them to bring them in line with `media-query-list-comma-space-*`.

We knew this had to be done at some point :) I think we were just putting off until someone actually requested it.

Feel free to take a stab at this @MadLittleMods . The [`media-query-list-comma-*`](https://github.com/stylelint/stylelint/blob/master/docs/rules.md#media-query-list) rules outline exactly what options are needed on what rules to achieve your desired code style.

Otherwise, we'll probably make a start on this once we've finished on the CLI branch.
